taui/src/components/form.js

```diff
@@ -136,15 +143,25 @@ class Form extends React.PureComponent<Props> {
     const {destination, network} = this.state
     const destinations: Array<AccountAddress> = userProfile ? userProfile.destinations : []
     const locations = destinations.map(d => {
+      // check for older profiles that contain destination purposes not saved with
+      // the 'TripPurpose.' prefix required by react-i18next
+      const translatablePurpose = d.purpose.startsWith('TripPurpose')
+        ? d.purpose
+        : 'TripPurpose.' + d.purpose
       return {
-        label: d.purpose + ': ' + d.location.label,
+        label: d.location.label,
         position: d.location.position,
-        value: d.location.position
+        value: translatablePurpose
       }
     })
-    const destinationFilterOptions = createDestinationsFilter(locations)
+    const locationsWithLabels = locations.map(loc => {
+      // generate temporary, translated destination labels menu options
+      return {...loc, label: t(loc.value) + ': ' + loc.label}
```

Comment:
If you make the other suggested changes then you'll also have to change this in order to generate the translation key correctly because it'll no longer be completely stored in the destination.
```suggestion
      return {...loc, label: t(`TripPurpose.${loc.value}`) + ': ' + loc.label}
```